src/app/home/home.page.ts

```diff
@@ -1,12 +1,282 @@
+    },
+
+    {
+      name: "Double Fibrations (or functors are categories)",
+      given_by: "me",
+      at: "ATCAT Seminar",
+      url_at: "https://www.mathstat.dal.ca/~fu/atcat/",
+      url_abs: "https://www.mathstat.dal.ca/~fu/atcat/",
+      url_video: "",
+      url_slides: "",
+      year: 2022,
+      date: "March 1",
+      proj: "Fibrations for Double Categories",
+    },
+
+    {
+      name: "Categories for the Learning Mathematician",
+      given_by: "me",
+      at: "Dalhousie Honours Seminar",
+      url_at: "https://www.dal.ca/faculty/science/math-stats/news-events/colloquium.html",
+      url_abs: "https://www.mathstat.dal.ca/~mszyld/abstract_szyld_honours_seminar.pdf",
+      url_video: "",
+      url_slides: "",
+      year: 2021,
+      date: "October 20",
+      proj: "Outreach",
+    },
+
+    {
+      name: "Minimal Bicategories of Fractions and Colimits of Bicategories",
+      given_by: "me",
+      at: "ATCAT Seminar",
+      url_at: "https://www.mathstat.dal.ca/~mamy/atcat/",
+      url_abs: "https://www.mathstat.dal.ca/~mamy/atcat/",
+      url_video: "",
+      url_slides: "",
+      year: 2021,
+      date: "October 12",
+      proj: "Minimal Fractions",
+    },
+
+  ];
+
+  public projects: Project[] = [{ name: "Minimal Fractions", description: "Minimal Fractions description", items: []},
+  { name: "Outreach", description: "Outreach description", items: []},
+  { name: "Fibrations for Double Categories", description: "Fibrations for Double Categories description", items: []},
+  { name: "Tannaka-Galois", description: "Tannaka-Galois description", items: []},
+  { name: "Model Bicategories", description: "Model Bicategories description", items: []},
+  { name: "Sigma Limits", description: "Sigma Limits description", items: []}];
+
+
+  public currentList: PersonalItem[] = this.articles;
+
+  public added_stuff = false;
+
+  constructor( ) {
+  }
+
+  segmentHasChanged(value) {
+    this.currentList = value.detail.value;
+    
+    // The first time value changes, create the list of items of each project:
+    if (this.added_stuff == false) {
+      for (let this_art of this.articles) {
+        for (let this_proj of this.projects) {
+          if (this_proj.name == this_art.proj ) {
+            this_proj.items.push( this_art );
+          }
+        }
+      }
+      for (let this_talk of this.talks) {
+        for (let this_proj of this.projects) {
+          if (this_proj.name == this_talk.proj ) {
+            this_proj.items.push( this_talk );
+          }
+        }
+      }
+
+      for (let this_proj of this.projects) {
+        this_proj.items = this_proj.items.sort((a, b) => (a.year > b.year) ? -1 : 1);
+      }
+
+      this.added_stuff = true;
+    }
+
+  }
+
+  // getDetails(item: Article | Project) {
+  //     return this.isArticle(item) ? [item.description, item.collab, item.published, item.url]: item.description;
+  // }
+
+  isArticle(item: Article | Project | Talk): item is Article {

```

Comment:
You can just make these straight static functions, add them to `DataDefinitions`.

Reply:
should these functions be part of a class?

Reply:
No they don't have to be, can be imported. They can be static functions in a file.